.github/workflows/main.yml

```diff
@@ -0,0 +1,209 @@
+# Continuous integration and pull request validation builds for the
+# master and maintenance branches.
+name: CI Build
+
+on:
+  push:
+    branches: [ master, maint/* ]
+  pull_request:
+    branches: [ master, maint/* ]
+
+env:
+  docker-registry: docker.pkg.github.com
+  docker-config-path: azure-pipelines/docker
+
+jobs:
+  # Build the docker container images that we will use for our Linux
+  # builds.  This will identify the last commit to the repository that
+  # updated the docker images, and try to download the image tagged with
+  # that sha.  If it does not exist, we'll do a docker build and push
+  # the image up to GitHub Packages for the actual CI/CD runs.  We tag
+  # with both the sha and "latest" so that the subsequent runs need not
+  # know the sha.
```

Comment:
That's awesome! Does this also work for PRs from forks that change Dockerfiles? Seems like it does based on below `github.repository` reference. :tada:

Reply:
I pushed up a change so that for CI builds (merges into master) that we will fetch the container if it exists for the current hash of the Dockerfiles, and it will build a new one and push it up if it doesn't.  For PR builds, we will just always build a new docker container andr run it for the build.

But this is too far in the other direction - what we really want is to use the container from GitHub Packages for PRs that don't touch the Dockerfiles, and build their own when they do.  I'll fix this up next...